sql/sql_parse.cc

```diff
@@ -5153,9 +5153,36 @@ mysql_execute_command(THD *thd)
       my_ok(thd);
     break;
   case SQLCOM_BACKUP_LOCK:
-    if (check_global_access(thd, RELOAD_ACL))
-      goto error;
-    /* first table is set for lock. For unlock the list is empty */
+    if (check_global_access(thd, RELOAD_ACL, true))
+    {
+      /* In case there is no global privilege, check DB privilege for BACK LOCK.*/
+      if (first_table)
+      {
+        if (check_single_table_access(thd, LOCK_TABLES_ACL, first_table, true))
+        {
+#ifndef NO_EMBEDDED_ACCESS_CHECKS
```

Comment:
this `#ifdef` could apply to the entire inner of the other `if` condition right? `check_single_table_access` isn't defined under `ifndef NO_EMBEDDED_ACCESS_CHECKS`.

Reply:
I guess so, yes.

Reply:
I guess so, yes.

Reply:
Does it make sense to wrap `#ifndef NO_EMBEDDED_ACCESS_CHECKS` in line `5158`, so for `embedded` basically it should only check global privilege,not database levels?

Reply:
wrap outside `check_global_access` which returns always true under `NO_EMBEDDED_ACCESS_CHECKS`. You've tried compiling in this mode?

Reply:
Yes, now I did. Wrapping outside `check_global_access` (no call for this function) doesn't work and raises an `DBUG_ASSERT(0)` in `Protocol::end_statement()`, so leaving as suggested before.

Reply:
ok.
 Remove the `goto error;` condition in the else branch, does that work?